File: scala/src/com/runtimeverification/ktest/kthings/aexp.scala
package com.runtimeverification.ktest.kthings

import com.runtimeverification.ktest.KThing

object aexp {
  class Aexp extends KThing
  case class AInt(int : KInt) extends Aexp
  case class AId(id : Id) extends Aexp
  case class Add(left: Aexp, right: Aexp) extends Aexp
  case class Div(left: Aexp, right: Aexp) extends Aexp
}

File: java.optimized/src/main/java/com/runtimeverification/ktest/nonterminals/Aexp.java
package com.runtimeverification.ktest.nonterminals;

import com.runtimeverification.ktest.KThing;

public abstract class Aexp {
    public static KThing aint(KThing value) {
        return new KThing(KThing.Type.AEXP_INT, value);
    }

    public static KThing id(KThing id) {
        return new KThing(KThing.Type.AEXP_ID, id);
    }

    public static KThing add(KThing first, KThing second) {
        return new KThing(KThing.Type.AEXP_ADD, first, second);
    }

    public static KThing div(KThing first, KThing second) {
        return new KThing(KThing.Type.AEXP_DIV, first, second);
    }
}
